### PR TITLE
Flush UI operation queue when animation is not running

### PR DESCRIPTION
**Pull request by piaskowyk**
## Summary

When animation is not running it is possible to enqueue UI update operation but never execute them.

| Before      | After |
| ----------- | ----------- |
| <video src="https://user-images.githubusercontent.com/36106620/230021472-c41aa77b-d955-4503-8273-a26e29dc7dea.mov" /> | <video src="https://user-images.githubusercontent.com/36106620/230021555-cbb62d7a-5f2e-4eb1-8d21-d3169aa1bfaf.mov" /> |

I tested it on the Fabric also.

<details>
<summary>code</summary>

```js
import Animated, {
  useSharedValue,
  useAnimatedStyle,
  useAnimatedProps,
  withTiming
} from 'react-native-reanimated';
import { View, Button } from 'react-native';
import React from 'react';

export default function AnimatedStyleUpdateExample(): React.ReactElement {
  const sv1 = useSharedValue(20);
  const sv2 = useSharedValue(20);
  const style1 = useAnimatedStyle(() => ({ width: sv1.value }));
  const style2 = useAnimatedStyle(() => ({ width: sv2.value }));

  return (
    <View
      style={{
        flex: 1,
        flexDirection: 'column',
        marginTop: 50
      }}>
      <Animated.View
        style={[
          { width: 100, height: 80, backgroundColor: 'orange'},
          style1,
        ]}
      />
      <Animated.View
        style={[
          { width: 100, height: 80, backgroundColor: 'black'},
          style2,
        ]}
      />
      <Button
        title={'set width for 1'}
        onPress={() => {
          sv1.value = withTiming(sv1.value === 20 ? 100 : 20);
        }}
      />
      <Button
        title={'set width for 2'}
        onPress={() => {
          sv2.value = sv2.value === 20 ? 100 : 20;
        }}
      />
      <Button
        title={'set width for 1 & 2'}
        onPress={() => {
          sv1.value = withTiming(sv1.value === 20 ? 100 : 20);
          sv2.value = sv2.value === 20 ? 100 : 20;
        }}
      />
    </View>
  );
}

```

</details>

### CODE REVIEW

File: ios/REANodesManager.h [hunk 1]
```diff
@@ -55,5 +55,5 @@ typedef void (^REAPerformOperations)();
                        trySynchronously:(BOOL)trySync;
 - (NSString *)obtainProp:(nonnull NSNumber *)viewTag propName:(nonnull NSString *)propName;
 #endif
-
+- (void)maybeFlushUIUpdatesQueue;
```
Comment (tomekzaw) [inline ios/REANodesManager.h:58]:
```suggestion
- (void)maybeFlushUIUpdatesQueue;

```

**Review by tomekzaw — APPROVED** [review]

**Review by kmagiera — APPROVED** [review]
LGTM 👍

**Review by tomekzaw — APPROVED** [review]
I can confirm this PR successfully resolves the issue with `useAnimatedSensor` ✅

**Comment by ranaavneet** [comment]
@piaskowyk @tomekzaw does this solve https://github.com/software-mansion/react-native-reanimated/issues/4231 as well?

**Comment by tomekzaw** [comment]
@ranaavneet unfortunately, probably not, but we're working on it

**Pull request merged**